core/src/main/scala/kafka/server/AbstractFetcherThread.scala

```diff
@@ -382,6 +382,11 @@ abstract class AbstractFetcherThread(name: String,
                     "that the partition is being moved")
                   partitionsWithError += topicPartition
 
+                case Errors.UNKNOWN_TOPIC_OR_PARTITION =>
+                  warn(s"Remote broker does not host the partition $topicPartition, which could indicate " +
+                    "that the partition is being created or deleted.")
```

Comment:
Should it be `info` if we think this is expected? That would still show in the logs.

Reply:
@ijuma I feel that `info` has an implicit implication that everything is "fine"; hence, I am concerned that it might be overlooked when investigating an issue.

The current broadcast-based approach in Kafka is just one way -- which arguably, has room for improvement -- of implementing state propagation for topic creation/deletion, and this error is a side-effect of it. Extended occurrence of this error can hamper the read/write availability as well as state consistency. Thus, even if `UNKNOWN_TOPIC_OR_PARTITION` is expected under the current design/implementation, it is definitely not desired.

What are your thoughts?

Reply:
I think that relying on warnings in logs to detect propagation issues is difficult. We have metrics to indicate issues with failed replica fetches and such. Are those enough for these cases?

Reply:
@ijuma Thanks for your reply!
We both agree that relying on warnings in logs to _detect_ propagation issues is not practical -- metrics indeed should be used in "detection" of such issues.
I believe the focus of this discussion isthe _investigation_ of detected issues.

Metrics typically fail to provide the level of details to examine the interaction of various events in a distributed environment. Hence, in case we need to look at the logs for figuring out such cases, I claim that being able to grep warns and see undesired events is useful -- as opposed to trying to browse info logs for bad/undesirable events.

Note that in `trunk`, such `UNKNOWN_TOPIC_OR_PARTITION` events are already printed in a stronger level (i.e. `error`).

Reply:
I still think that `info` would be a better debug level for this. But let's get a second opinion from @junrao or @hachikuji.

Reply:
I don't have a strong opinion on this one. I guess we would most likely see this in startup cases where the remote broker may not have gotten the full metadata yet. If we fixed that common case, then I think there would be a strong case to change this to warn. Otherwise, we'd just learn to ignore them in spite of the higher level.

Reply:
If a broker is restarted, it typically doesn't have any leader replicas and won't receive follower requests. So, this is mostly related to topic creation/deletion. When there is topic creation/deletion, receiving transient UNKNOWN_TOPIC_OR_PARTITION is expected. But sustained UNKNOWN_TOPIC_OR_PARTITION is unexpected. So, perhaps we could keep the logging at WARN, but tweak the message a bit to sth like "Receiving UNKNOWN_TOPIC_OR_PARTITION from the leader for partition XXXX. This can happen transiently if the partition is being created or deleted. However, this is unexpected if it sustains."

Reply:
@junrao @hachikuji @ijuma Thanks for the review!
I updated the message as per Jun's recommendation -- please let me know if further changes needed.